src/http/client.cc

```diff
@@ -312,6 +312,15 @@ auto client::with_connection(Fn&& fn) {
     });
 }
 
+struct transport_error : public std::exception {
```

Comment:
how about renaming it to `transient_transport_error` ?

Reply:
also, nit, this `exception` is part of the internal implementation of the retry machinery. it will never be caught as an `exception`. so no need to inherit from `exception` and no need to override `what`.

Reply:
Not quite. The `do_make_request()` is called two times by `make_request` and the `transport_error` is handled only by the first call. If the second one ends up with the same exception, it will be propagated back to the caller, so it must be the `std::exception` with `what()` too